src/Workspaces/Core/Portable/Shared/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -90,14 +90,6 @@ internal static partial class ITypeSymbolExtensions
             var constructedInterfaces = typeSymbol.AllInterfaces.Where(i =>
                 SymbolEquivalenceComparer.Instance.Equals(i.OriginalDefinition, originalInterfaceType));
 
-            // Attempt to find and cache the compilation for the types we're searching for implementations in. This will
-            // be used to try to check equivalence when forwarded types are involved when we call into
-            // OriginalSymbolsMatch.  This is just an optimization as OriginalSymbolsMatch will also do this.  However,
-            // we may be making many different calls to OriginalSymbolsMatch below and there's no need to do this
-            // multiple times.
-            SymbolFinder.TryGetCompilation(typeSymbol, solution, out var typeSymbolCompilation);
-            SymbolFinder.TryGetCompilation(interfaceMember, solution, out var interfaceMemberCompilation);
```

Comment:
this was the suspect code.  core issues were twofold:

1. we're comparing lots of differnet symbols (included constructed symbols).  it's not necessarily the case that all of these should use the same compilation's assembly.  it was much easier to just grab the right compilation per symbol we're comparing, and just do the caching at that level to save on perf costs so we don't do that multiple times.
2. TryGetCompilation is actually somewhat weird as it will sometimes do the following:
  a. if you have a symbol from project A that you are seeing through project B.  it will attempt to get the compilation for 'A", not B, even though A will have retargetted portoins of it.  This basically just leads to an overall inconsistency where none of teh questions about forwarded types even makes sense.